---
title: "RNAseq of SRP350860 (Rhizoclosmatium globosum JEL800)"
author: "Kelterborn"
date: "2024-06-24"
output:
  github_document:
    html_preview: false
    toc: true
    toc_depth: 2
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
    rmarkdown::render(input_file,output_file= 'Readme.md')
        })
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=150)

```


RNA-Seq of Rhizoclosmatium globosum JEL800

Raw files
https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP350860

# 0) Preparation
## -R_libraries
```{r r_libraries, echo=FALSE}

# BiocManager::install()

library(GenomicFeatures)
library(tximeta)
library(tximport)
library(DESeq2)
library(curl)
library(AnnotationHub)
library(patchwork)
library(PCAtools)
library(stringr)
library(R.utils)
library(RColorBrewer)
library(sessioninfo)
library(data.table)
library(plyr)
library(tidyverse)

library(EnhancedVolcano)
library(pheatmap)
library(writexl)
library(biomaRt)
library(ape)
library(kableExtra)
library(knitr)

library(stringr)
library(R.utils)
library(RColorBrewer)

library(sessioninfo)
library(data.table)
library(plyr)
library(tidyverse)
library(tximeta)
library(tximport)
library(curl)
library(DESeq2)

library(SummarizedExperiment)
library(GenomicRanges)
library(ape)

library(viridis)

library(ggpubr)
library(vsn)
library(stringr)
library(R.utils)
library(RColorBrewer)

library(VennDiagram)
library(GenomicFeatures)
library(phylotools)
library(ape)

library(writexl)



```

## -R_folders
```{r r_folders, include = FALSE}

# Linux Workstation
ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
    ifelse(Sys.info()["sysname"]== "Darwin",
       s <- "/mnt/s",
       s <- "S:"))
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/Rhihy1",sep="/")
# list.files(dir) %>% head()
gitdir <- paste(dir,"git_Rhihy1",sep="/")
# list.files(gitdir) %>% head()

pub_figures <- paste(gitdir,"pub_figures",sep="/")

quantfolder <- paste(dir,"quants",sep="/")
fastq <- paste(dir,"fastq",sep="/")
fastaPath <- paste(dir,"genomic/Rhihy1_AssemblyScaffolds_Repeatmasked.fasta.gz",sep="/")
gtfPath <- paste(dir,"genomic/Rhihy1_GeneCatalog_genes_20151219.gff.gz",sep="/")
indexDir <- paste(dir,"index/rhihy1_v1_index",sep="/")
quantfolder <- paste(dir,"quants",sep="/")

```

## -R functions
```{r f_get_results}

```


## -Unix packages
```{bash unix_packages, eval=FALSE}

mamba create -n salmon salmon fastqc multiqc sra-tools
mamba activate salmon

```

## -Unix folders
```{bash unix_folders, eval=FALSE}
dir="/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Rhihy1"
fastq=$(echo $dir/fastq)
index=$(echo $dir/index/rhihy1_v1_index)
genomic=$(echo $dir/genomic)

```


## -Download data
### -FASTQ files
https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP350860

Download "Accession List" as text file.
Download via NCBI SRA tools (unix)
Note: alternative to fast-dump: faster-dump or parallel-fastq-dump https://github.com/ncbi/sra-tools/wiki/HowTo:-fasterq-dump

```{bash download_sra, eval=FALSE}
cd $fastq
srr=$(cat SRR_Acc_List.txt)
prefetch -v $srr
for fn in $srr; do fastq-dump --gzip --outdir . --split-files $fn; done


```

### -Rhihy1 genomic data

Download genomic datra from: https://mycocosm.jgi.doe.gov/Rhihy1/Rhihy1.home.html
-> https://genome.jgi.doe.gov/portal/pages/dynamicOrganismDownload.jsf?organism=Rhihy1

Rhihy1_AssemblyScaffolds_Repeatmasked.fasta.gz
Rhihy1_EST_20151219_cluster_consensi.fasta.gz
Rhihy1_FilteredModels1_deflines.gff3.gz

#### -fix transcriptome file
Adapt Transcriptome names to GFF file

```{r fix_transcriptome, eval=FALSE}
# Gene catalogue
gtfdata <- readLines(file.path(paste(dir,"genomic/Rhihy1_GeneCatalog_genes_20151219.gff.gz",sep="/")))
head(gtfdata, n=10)

# Transcriptome
transcripts_path <- file.path(paste(dir,"genomic/Rhihy1_GeneCatalog_transcripts_20151219.nt.fasta.gz",sep="/"))
transcripts <- readLines(transcripts_path)
head(transcripts, n=10)[2] %>% str_length()

# extract seq names
old_name <- get.fasta.name(transcripts_path)
length(old_name)

transcriptId <- str_split(old_name,pattern = "\\|", simplify = T)[,3]
length(transcriptId)

transcript_name <- str_split(old_name,pattern = "\\|", simplify = T)[,4]
length(transcript_name)

# generate new name
new_name <- paste(transcript_name, " transcriptId:",transcriptId,sep="")

# replace old names
transcripts_path_new <- "/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Rhihy1/genomic/Rhihy1_GeneCatalog_transcripts_20151219.nt.fixed.fasta"
ref2 <- data.frame(old_name, new_name)
rename.fasta(infile = transcripts_path, ref_table = ref2, outfile = transcripts_path_new)
unlink("transcripts_path")
unlink("transcripts_path_new")

transcripts <- readLines(transcripts_path_new)
head(transcripts, n=10)

# -> wrong line breaks!
```

#### -add line breaks
install fastx-toolkit

```{bash fastx, eval=FALSE}
# fasta_formatter -h
#	usage: fasta_formatter [-h] [-i INFILE] [-o OUTFILE] [-w N] [-t] [-e]


fasta_formatter -i Rhihy1_GeneCatalog_transcripts_20151219.nt.fixed.fasta -o Rhihy1_GeneCatalog_transcripts_20151219.nt.fixed.lines.fasta -w 70

gzip -k Rhihy1_GeneCatalog_transcripts_20151219.nt.fixed.lines.fasta

```

```{r check_fastx}

readLines(paste(dir,"/genomic/Rhihy1_GeneCatalog_transcripts_20151219.nt.fixed.lines.fasta.gz",sep="")) %>% head(n=10)

```




# 1) Mapping 
## -Make index 
```{bash make_index, eval=FALSE}

cd $genomic

# create list of chromosomes -->
grep "^>" <(gunzip -c Rhihy1_AssemblyScaffolds_Repeatmasked.fasta.gz) | cut -d " " -f 1 > decoys_rhihy1_v1.txt

# the created .txt contains ">" in front of all chromosomes, use to remove:
sed -i.bak -e 's/>//g' decoys_rhihy1_v1.txt

# check that ">" is removed
nano decoys_rhihy1_v1.txt
# to exit vim: press Esc -> :q -> Enter

# Combine the different files
cat Rhihy1_GeneCatalog_transcripts_20151219.nt.fixed.lines.fasta.gz Rhihy1_AssemblyScaffolds_Repeatmasked.fasta.gz  > gentrome_rhihy1_v1.fa.gz

# make index with salmon
# the number behind -p defines the number of cores used to index. Currently (04/2024) 10 is the maximum... this is weird. If you use more, an index is generated but some data is missing, leading to downstream problems.

mamba activate salmon

cd $genomic
salmon index -t gentrome_rhihy1_v1.fa.gz -d decoys_rhihy1_v1.txt -p 10 -i $index


```


## -Quantification
```{bash mapping, eval=FALSE}

mamba activate salmon

fastqfiles=$(ls $fastq/*_1.fastq.gz)

echo "${fastqfiles:0:200}"

for fn in $fastqfiles;
do
bfn=$(echo "`basename ${fn}`");
samp=${bfn%"_1.fastq.gz"};
R1=$fn;
R2=$(echo "$R1" | sed 's/_1/_2/');
echo "Processing Sample: $samp";
test -f $R1 && echo "--> File: $R1 exists"
test -f $R2 && echo "--> File: $R2 exists"
outquant=${dir}/quants/${samp}_quant
echo $outquant
echo $index

salmon quant -i $index -l A \
 -1 $R1 \
 -2 $R2 \
 -p 25 --validateMappings --gcBias -o $outquant

done

```

# 2) Differential gene expression analysis
## -Sample table
```{r sample_table}

# generate file list from actual mapping files
files={}
for (i in list.dirs(path = quantfolder, full.names = TRUE, recursive = FALSE)) {
files <- c(files,paste(i,"/quant.sf",sep=""))
# print(head(read.table(files[length(files)], header=T)))
}
readLines(files[length(files)]) %>% head()

coldata <- data.frame(files=files,
                      run=list.files(quantfolder) %>% str_remove(.,"_quant"))
coldata

# get sample infos from metadata table
accession_table <- read_csv(file=paste(dir,"/fastq/SraRunTable.txt",sep="")) %>% data.frame()
colnames(accession_table)

# combine metadata with mapping files
coldata <- cbind(coldata,
                 accession_table[,c("Run","Sample.Name","life_stage","strain","SRA.Study","BioProject","BioSample","Experiment")])

# Define sample names
coldata$names <- coldata$Sample.Name

# Make readable condition and order
coldata$condition <- coldata$life_stage %>%
  str_replace(pattern="mature thallus", replacement = "mat_thallus") %>% factor(levels=c("zoospore","germling","immat_thallus","mat_thallus"))

```


## -Tximeta
Summarize all transcript counts to gene level (no gene infos yet... transcripts = genes )
```{r tximeta}
# Make manual linked transcriptome

# set paths
file.exists(indexDir, fastaPath, gtfPath)

quant <- read.delim(files[1])
gtfdata <- read.gff(gtfPath, na.strings = c(".", "?"), GFF3 = TRUE)

head(quant, n=10)
head(gtfdata, n=10)

tx_names <- quant$Name
tx_names %>% length()
tx_names %>% unique() %>% length()


str_count(quant,pattern="gm1.1_g") %>% table()
str_count(gtfdata,pattern="gm1.1_g") %>% table()

str_count(quant,pattern="CE99926_618") %>% table()
str_count(gtfdata,pattern="CE99926_618") %>% table()

# # clear bfc
# bfcloc <- getTximetaBFC()
# bfc <- BiocFileCache(bfcloc)
# bfcquery(bfc, "")
# bfcquery(bfc, "Rhihy1")
# 
# bfcremove(bfc, bfcquery(bfc, "Rhihy1")$rid)
# bfcremove(bfc, bfcquery(bfc, "linkedTxomeTbl")$rid)

makeLinkedTxome(indexDir=indexDir,
                 source="MycoCosm",
                 organism="Rhizoclosmatium globosum",
                 release="v1",
                 genome="Rhihy1",
                 fasta=fastaPath,
                 gtf=gtfPath,
                 write=FALSE)

# use tximeta on coldata and save as summarized experiment se






# Use tx2gene

tx2gene <- data.frame(TXNAME=tx_names,
                      GENEID=tx_names)
files <- coldata[,"files"]
names(files) <- coldata[,"names"]

txi <- tximport(files=files, type = "salmon", tx2gene = tx2gene)
head(txi$counts)

sampleTable <- coldata
rownames(sampleTable) <- colnames(txi$counts)
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~condition)

```

### -Check mapping rates
```{r se_QC, fig.show='hold', out.width="50%", eval=FALSE}
# Plot data:
# The scale of the y axis can be changed by setting ylim = c(y1,y2) values 

barplot(metadata(se)[["quantInfo"]]$percent_mapped, main="Mapping Rate, all samples", names = colnames(se), las = 2, cex.names = 0.5, ylim = c(0,80))
barplot(metadata(se)[["quantInfo"]]$num_processed/1000000, main="Mio. Reads, all samples", names = colnames(se), las = 2, cex.names = 0.5, ylim = c(0,30))
```

## -DESeq2
```{r Deseq2}
dds <- DESeq(dds) 

# Plot example counts
plotCounts(dds,gene = "CE559232_26255", intgroup = "condition")

# Make results
resultsNames(dds)
res_germling_vs_zoospore <- results(dds, contrast = c("condition","germling","zoospore"))
res_immat_thallus_vs_germling <- results(dds, contrast = c("condition","immat_thallus","germling"))
res_mat_thallus_vs_immat_thallus <- results(dds, contrast = c("condition","mat_thallus","immat_thallus"))
res_zoospore_vs_mat_thallus <- results(dds, contrast = c("condition","zoospore","mat_thallus"))


```

# Export data
```{r export, eval=FALSE}

res_exp <- data.frame(
  # general
  "name" = rownames(dds),
 # results res_germling_vs_zoospore
 "L2FC.germling_vs_zoospore" = res_germling_vs_zoospore$log2FoldChange,
 "pvalue.germling_vs_zoospore" = res_germling_vs_zoospore$pvalue,
 "padj.germling_vs_zoospore" = res_germling_vs_zoospore$padj,
  # results res_immat_thallus_vs_germling
 "L2FC.res_immat_thallus_vs_germling" = res_immat_thallus_vs_germling$log2FoldChange,
 "pvalue.res_immat_thallus_vs_germling" = res_immat_thallus_vs_germling$pvalue,
 "padj.res_immat_thallus_vs_germling" = res_immat_thallus_vs_germling$padj,
  # results res_mat_thallus_vs_immat_thallus
 "L2FC.res_mat_thallus_vs_immat_thallus" = res_mat_thallus_vs_immat_thallus$log2FoldChange,
 "pvalue.res_mat_thallus_vs_immat_thallus" = res_mat_thallus_vs_immat_thallus$pvalue,
 "padj.res_mat_thallus_vs_immat_thallus" = res_mat_thallus_vs_immat_thallus$padj,
  # results res_zoospore_vs_mat_thallus
 "L2FC.res_zoospore_vs_mat_thallus" = res_zoospore_vs_mat_thallus$log2FoldChange,
 "pvalue.res_zoospore_vs_mat_thallus" = res_zoospore_vs_mat_thallus$pvalue,
 "padj.res_zoospore_vs_mat_thallus" = res_zoospore_vs_mat_thallus$padj,
  counts(dds, normalized=TRUE)[,c("ZSP_1","ZSP_2","ZSP_3", "GER_1", "GER_2", "GER_3","IMA_1", "IMA_2", "IMA_3","MAT_1", "MAT_2", "MAT_3")]
 )
res_exp

write_xlsx(data.frame(res_exp),
           paste(gitdir,"2024_06 Rhihy1 rna-seq.xlsx",sep="/"))


```

